src/machines/settingsMachine.ts

```diff
@@ -37,17 +45,16 @@ export const settingsMachine = setup({
     setClientSideSceneUnits: () => {},
     persistSettings: () => {},
     resetSettings: assign(({ context, event }) => {
-      if (!('defaultDirectory' in event)) return {}
-      // Reset everything except onboarding status,
-      // which should be preserved
-      const newSettings = createSettings()
-      if (context.app.onboardingStatus.user) {
-        newSettings.app.onboardingStatus.user =
-          context.app.onboardingStatus.user
-      }
-      // We instead pass in the default directory since it's asynchronous
-      // to re-initialize, and that can be done by the caller.
-      newSettings.app.projectDirectory.default = event.defaultDirectory
+      if (!('level' in event)) return {}
```

Comment:
That's a neat one. Why not `if (!event.level)` ?

Reply:
It's the same in this case! I could totally do that, and it's probably a hair better.

In other type narrowing for these XState events we'll have `data` objects and have to narrow based on the keys _within_ the `data` object, that may or may not exist. TS will yell at you if you try to type narrow in that instance with something like `(!event.data?.innerKey)`, so you have to narrow with something like `(!('data' in event && event.data.innerKey))`. So I've just matched that idiom.

Reply:
From what I understand all data in XState (5) is strongly typed, so I'd be surprised if we ever need `in` but please ping me so I can collect some examples for the ol' brain vault :D